src/sentry/release_health/metrics_sessions_v2.py

```diff
@@ -425,13 +435,79 @@ def run_sessions_query(
     project_ids = filter_keys.pop("project_id")
     assert not filter_keys
 
-    orderby = _parse_orderby(query, fields)
-    if orderby is None:
-        # We only return the top-N groups, based on the first field that is being
-        # queried, assuming that those are the most relevant to the user.
-        # In a future iteration we might expose an `orderBy` query parameter.
-        primary_metric_field = _get_primary_field(list(fields.values()), query.raw_groupby)
-        orderby = OrderBy(primary_metric_field, Direction.DESC)
+    ordered_preflight_filters: Dict[GroupByFieldName, Sequence[str]] = {}
+    try:
+        orderby = _parse_orderby(query, fields)
+    except NonPreflightOrderByException as exc:
+        # We hit this branch when we suspect that the orderBy columns is one of the virtual
+        # columns like `release.timestamp` that require a preflight query to be run, and so we
+        # check here if it is one of the supported preflight query columns and if so we run the
+        # preflight query. Otherwise we re-raise the exception
+        raw_orderby = query.raw_orderby[0]
+        if raw_orderby[0] == "-":
+            raw_orderby = raw_orderby[1:]
+            direction = Direction.DESC
+        else:
+            direction = Direction.ASC
+
+        if raw_orderby not in PREFLIGHT_QUERY_COLUMNS:
+            raise exc
+
+        preflight_query_conditions = {
+            "orderby_field": raw_orderby,
+            "direction": direction,
+            "org_id": org_id,
+            "project_ids": project_ids,
+            "limit": query.limit,
+        }
+
+        # For preflight queries, we need to evaluate environment conditions because these might
+        # be used in the preflight query. Example when we sort by `-release.timestamp`, and when
+        # we have environment filters applied to the query, then we need to include the
+        # environment filters otherwise we might end up with metrics queries filters that do not
+        # belong to the same environment
+        environment_conditions = []
+        for condition in where:
+            preflight_query_condition = _get_filters_for_preflight_query_condition(
+                tag_name="environment", condition=condition
+            )
+            if preflight_query_condition != (None, None):
+                environment_conditions += [preflight_query_condition]
```

Comment:
if i understand correctly this line are you trying to do something like:
```
 environment_conditions.append(preflight_query_condition)
```
?

Reply:
yes

Reply:
so my suggestion is just easy to read and based on this SO [answer](https://stackoverflow.com/questions/725782/in-python-what-is-the-difference-between-append-and) 2x faster ;)

Reply:
Oh nice ! TIL